redis/src/cluster_async/mod.rs

```diff
@@ -377,17 +380,67 @@ struct RequestInfo<C> {
 impl<C> RequestInfo<C> {
     fn set_redirect(&mut self, redirect: Option<Redirect>) {
         if let Some(redirect) = redirect {
-            let redirect = InternalSingleNodeRouting::Redirect(redirect);
             match &mut self.cmd {
-                CmdArg::Cmd { routing, .. } => {
-                    *routing = redirect.into();
-                }
+                CmdArg::Cmd { routing, .. } => match routing {
+                    InternalRoutingInfo::SingleNode(route) => {
+                        let redirect = InternalSingleNodeRouting::Redirect {
+                            redirect,
+                            previous_routing: Box::new(std::mem::replace(
+                                route,
+                                InternalSingleNodeRouting::Random,
```

Comment:
I don't understand why do you replace 'route' with Random node, when is it going to be used?
please add a comment explaining the logic

Reply:
https://doc.rust-lang.org/std/mem/fn.replace.html